tests/unit/objects/test_cylinder.py

```diff
@@ -321,3 +322,35 @@ def test_to_points(cylinder, n_along_axis, n_angles, points_expected):
     points_unique = Points(array_rounded).unique()
 
     assert points_unique.is_close(points_expected)
+
+
+@pytest.mark.parametrize(
+    ("points", "message_expected"),
+    [
+        ([[1, 0], [-1, 0], [0, 1]], "The points must be 3D."),
+        ([[2, 0, 1], [-2, 0, -3]], "There must be at least 6 points."),
+        ([[0, 0, 0], [1, 1, 1], [2, 2, 2], [3, 3, 3], [4, 4, 4], [5, 5, 5]], "The points must not be collinear."),
+        ([[0, 0, 1], [1, 1, 1], [2, 1, 1], [3, 3, 1], [4, 4, 1], [5, 5, 1]], "The points must not be coplanar."),
+    ],
+)
+def test_best_fit_failure(points, message_expected):
+
+    with pytest.raises(ValueError, match=message_expected):
+        Cylinder.best_fit(points)
+
+
+@pytest.mark.parametrize(
+    ("points", "point_expected", "vector_expected", "radius_expected"),
+    [
+        ([[2, 0, 0], [0, 2, 0], [0, -2, 0], [2, 0, 4], [0, 2, 4], [0, -2, 4]], [0, 0, 0], [0, 0, 4], 2.0),
+        ([[-2, 0, 1], [-2, 1, 0], [-2, -1, 0], [3, 0, 1], [3, 1, 0], [3, -1, 0]], [-2, 0, 0], [5, 0, 0], 1.0),
+        ([[-3, 3, 0], [0, 3, 3], [0, 3, -3], [-3, -12, 0], [0, -12, 3], [0, -12, -3]], [0, 3, 0], [0, -15, 0], 3.0),
+    ],
+)
+def test_best_fit(points, point_expected, vector_expected, radius_expected):
+
+    cylinder = Cylinder.best_fit(points)
+
+    assert cylinder.point.is_close(point_expected, abs_tol=1e-9)
+    assert cylinder.vector.is_close(vector_expected, abs_tol=1e-9)
+    assert math.isclose(cylinder.radius, radius_expected, abs_tol=1e-9)
```

Comment:
Are these tolerances needed for the tests to pass?

Reply:
I have removed the radius tolerance. The others are necessary for the tests to pass.